src/Partials/AlignBoxButton.vala

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2019 Alecaddd (http://alecaddd.com)
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program; if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA
+ *
+ * Authored by: Giacomo "giacomoalbe" Alberini <giacomoalbe@gmail.com>
+ */
+public class Akira.Partials.AlignBoxButton : Gtk.Grid {
+    public string icon_name;
+    public string action;
+
+    public Gtk.IconSize icon_size;
+
+    private Gtk.Button button;
+    private Gtk.Image image;
+
+    public signal void triggered (Akira.Partials.AlignBoxButton emitter);
+
+    public AlignBoxButton (
+        string action,
+        string icon_name,
+        string title) {
+        this.action = action;
```

Comment:
We should also use gobject style constructors here :) 

- First, Make the properties we want to set by default like the icon_name, action and title as `public string icon_name { get; construct; }`
- Then, set the values using the `Object()` call, as you did in the AlignItemsPanel
- And finaly, put the rest of the code building the object inside of a `construct {` block :)